core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java

```diff
@@ -73,10 +97,32 @@ public void initializeLogging(LogConfig config) {
         InitialConfigurator.DELAYED_HANDLER.setHandlers(handlers.toArray(EmbeddedConfigurator.NO_HANDLERS));
     }
 
+    private boolean hasColorSupport() {
+
+        if (IS_WINDOWS) {
+            if (!(IS_CON_EMU_ANSI || IS_CYGWIN || IS_MINGW_XTERM)) {
+                // On Windows without a known good emulator
+                // TODO: optimally we would check if Win32 getConsoleMode has
+                // ENABLE_VIRTUAL_TERMINAL_PROCESSING enabled or enable it via 
+                // setConsoleMode.
+                // For now we turn it off to not generate noisy output for most
+                // users.
+                return false;
+            } else {
+                // Must be on some Unix variant or ANSI-enabled windows terminal...
+                return true;
```

Comment:
This should also return `System.console() != null` IMO.  Otherwise if you pipe through e.g. `more`, you'll get lots of garbage.

Reply:
then cygwin won't work.

Reply:
but yeah - just tested and piping goes bad...dammit.

Reply:
It's OK if cygwin doesn't work IMO.  They can enable colors manually.

Reply:
note: `mvn compile | less` on cygwin results in color dumps too.

Reply:
cygwin will likely soon stop returning `null` for console: https://sourceware.org/git/?p=newlib-cygwin.git;a=commit;h=169d65a5774acc76ce3f3feeedcbae7405aa9b57